front_end/entrypoints/inspector_main/DataBindingModelsPanel.ts

```diff
@@ -146,305 +146,511 @@ export class DataBindingModelsPanelView extends UI.ThrottledWidget.ThrottledWidg
         return dataBindingModelsViewInstance;
     }
 
+    addAutoRefreshInterval() {
+        if (this._watchInterval) return;
+
+        this._inputIntervalWrapper?.classList.remove('hidden');
+
+        this._watchInterval = setInterval(() => this.update(), this._watchIntervalValue.get())
+    }
+
+    clearWatchInterval() {
+        if (this._watchInterval) {
+            this._inputIntervalWrapper?.classList.add('hidden');
+            clearInterval(this._watchInterval);
+            this._watchInterval = null;
+        }
+    }
+
+    handleAutoUpdateSetting() {
+        if (this._autoUpdateBindModelsSetting.get()) {
+            this.addAutoRefreshInterval();
+        } else {
+            this.clearWatchInterval();
+        }
+    }
+
+    createToolbar() {
+        this._toolbarItems = [];
+
+        const refreshButton =
+            new UI.Toolbar.ToolbarButton(i18nString(UIStrings.refreshModels), 'largeicon-refresh');
+        refreshButton.addEventListener(UI.Toolbar.ToolbarButton.Events.Click, this.update, this);
+        this._toolbarItems.push(refreshButton);
+
+        this._toolbarItems.push(new UI.Toolbar.ToolbarSeparator());
+
+        const loadModelsIcon = new UI.Toolbar.ToolbarButton(i18nString(UIStrings.loadModels), 'largeicon-load');
+        loadModelsIcon.addEventListener(UI.Toolbar.ToolbarButton.Events.Click, this._selectFileToLoad.bind(this));
+        const exportModelsIcon = new UI.Toolbar.ToolbarButton(i18nString(UIStrings.exportModels), 'largeicon-download');
+        exportModelsIcon.addEventListener(UI.Toolbar.ToolbarButton.Events.Click, this._saveToFile.bind(this));
+
+        this._toolbarItems.push(loadModelsIcon);
+        this._toolbarItems.push(exportModelsIcon);
+
+        this._toolbarItems.push(new UI.Toolbar.ToolbarSeparator());
+
+        // Disable for now
+        // const addButton = new UI.Toolbar.ToolbarButton(i18nString(UIStrings.addModel), 'largeicon-add');
+        // addButton.addEventListener(UI.Toolbar.ToolbarButton.Events.Click, this._addButtonClicked, this);
+        // this._toolbarItems.push(addButton);
+
+        this._autoUpdateBindModelsSetting.addChangeListener(this.handleAutoUpdateSetting.bind(this));
+
+        this._toolbarItems.push(new UI.Toolbar.ToolbarSettingCheckbox(
+            this._autoUpdateBindModelsSetting, i18nString(UIStrings.autoUpdateBindModels), i18nString(UIStrings.watchForModelChanges)));
+
+        this._intervalInput = document.createElement('input') as HTMLInputElement;
+        this._intervalInput.setAttribute('type', 'number');
+        this._intervalInput.setAttribute('inputmode', 'numeric')
+        this._intervalInput.setAttribute('pattern', '\d*')
+        this._intervalInput.classList.add('update-models-interval-input');
+        this._intervalInput.addEventListener('blur', this.handleWatchIntervalChange.bind(this));
```

Comment:
Maybe we can shorten it like this?
```suggestion
        this._intervalInput = Object.assign(document.createElement('input') as HTMLInputElement, {
            type: 'number',
            inputMode: 'numeric',
            pattern: '\\d*',
            className: 'update-models-interval-input',
        })
        this._intervalInput.addEventListener('blur', this.handleWatchIntervalChange.bind(this));
```

Reply:
Nice, thanks! Fixed!